modules/engine/src/main/scala/observe/engine/Step.scala

```diff
@@ -24,16 +25,17 @@ final case class Step[F[_]](
 
 object Step {
 
-  final case class BreakpointMark(self: Boolean) extends AnyVal
-
-  final case class SkipMark(self: Boolean) extends AnyVal
-
-  final case class Skipped(self: Boolean) extends AnyVal
+  object BreakpointMark extends NewType[Boolean]
+  type BreakpointMark = BreakpointMark.Type
+  object SkipMark extends NewType[Boolean]
```

Comment:
I think we decided not to have skipped steps anymore. Maybe we can remove the logic? It can be another PR.

Reply:
Ok, I'll do that, there is some leftover code on the engine about them